File: McbEdu.Mentorias.ShopDemo.Services/Customers/Interfaces/ICustomerService.cs
using McbEdu.Mentorias.DesignPatterns.NotificationPattern;
using McbEdu.Mentorias.ShopDemo.Domain.Contexts.CustomerContext.DTO;
using McbEdu.Mentorias.ShopDemo.Services.Customers.Inputs;

namespace McbEdu.Mentorias.ShopDemo.Services.Customers.Interfaces;

public interface ICustomerService
{
    Task<(bool HasExecuted, List<NotificationItem> Notifications)> ImportCustomerAsync(ImportCustomerServiceInput input);
    Task<(bool HasExecuted, List<NotificationItem> Notifications, List<Customer> Customers)> GetCustomerNoFilterAsync(GetCustomerServiceInput input);
    Task<(bool HasExecuted, List<NotificationItem> Notifications, List<Customer> Customers)> GetCustomerFilteredByEmailAsync(GetCustomerServiceFilteredByEmailInput input);
    Task<(bool HasExecuted, List<NotificationItem> Notifications, List<Customer> Customers)> GetCustomerFilteredByNameOrSurnameAsync(GetCustomerServiceFilteredByNameOrSurnameInput input);
    Task<(bool HasExecuted, List<NotificationItem> Notifications, List<Customer> Customers)> GetCustomerFilteredByRangeBirthDateAsync(GetCustomerServiceFilteredByRangeBirthDateInput input);
}

File: src/main/java/com/mcbmentorias/shopdemo/domain/services/interfaces/ICustomerService.java
package com.mcbmentorias.shopdemo.domain.services.interfaces;

import com.mcbmentorias.shopdemo.domain.entities.customer.Customer;
import com.mcbmentorias.shopdemo.domain.entities.customer.inputs.ImportNewCustomerInput;

public interface ICustomerService extends IService<Customer> {

    Boolean importCustomer(final ImportNewCustomerInput input);

    Boolean checkIfCustomerExistsByEmail(final String email);

    Customer getByEmail(final String email);

    Boolean createImportCustomerOrNotifyDifferences(final ImportNewCustomerInput inputModel);
}
